app/Http/Controllers/ProjectController.php

```diff
@@ -91,4 +96,34 @@ public function delete($id)
         // TODO: Notice of success?
         return redirect('/projectfinder');
     }
+
+    /**
+     * Adds the comment to the given project as long as the user is logged in
+     *
+     * @param $request = user entered text, $project project_id for lookup
+     * @return back to same page
+     */
+    public function addComment(Request $request, Project $project)
+    {
+        if(Auth::guest())                                               //Checks if user is not logged in
+        {
+            Session::flash('guestComment', 'You must be logged in to submit a comment');
+            return back();
+        }
+
+
+        $this->validate($request, [
+            'comment' => 'required|min:6|max:256|regex:/^(?=.*[a-zA-Z])([a-zA-Z0-9 -+_!@#$%^&*.,?]+)$/'
```

Comment:
Looks like there's a rule for this: https://laravel.com/docs/5.1/validation#rule-alpha-dash